gulpfile.js

```diff
@@ -62,12 +62,24 @@ gulp.task('tslint', function () {
 });
 
 function test() {
+    // Defaults
+    var pattern = "extensionContext";
+    var invert = true;
+
+    // Check if arguments were passed
+    var patternIndex = process.argv.indexOf("--pattern");
+    if (patternIndex > -1 && (patternIndex + 1) < process.argv.length) {
```

Comment:
This is misleading. I would either use `patternIndex` for both comparisons or `patternIndex + 1`.